app/uk/gov/hmrc/slacknotifications/model/SlackMessage.scala

```diff
@@ -37,15 +37,41 @@ case class Attachment(
                        footer: Option[String],
                        footer_icon: Option[String],
                        ts: Option[Int]
-                     )
+                     ) {
+
+  def getFields: Array[String] = {
+    val baseFields = Array(
+      this.fallback,
+      this.color,
+      this.pretext,
+      this.author_name,
+      this.author_link,
+      this.author_icon,
+      this.title,
+      this.title,
+      this.text,
+      this.image_url,
+      this.thumb_url,
+      this.footer,
+      this.footer_icon
+    )
+
+    val additionalFields: Array[String] = if(this.fields.isDefined) this.fields.map(fieldSeq => fieldSeq.flatMap(_.fields).toArray).get else Array[String]()
```

Comment:
the `if(option.isDefined) option.get` is not very idiomatic in scala. Could you maybe `.getOrElse`, `.fold` or pattern match instead?

Reply:
Done :+1: